go/src/lib.rs

```diff
@@ -170,11 +187,166 @@ pub unsafe extern "C" fn close_client(client_adapter_ptr: *const c_void) {
+
+    let mut client_clone = client_adapter.client.clone();
+    client_adapter.runtime.spawn(async move {
+        let mut cmd = command_type.get_command().unwrap(); // TODO check cmd
+                                                           //print!("{:?}", cmd.args);
+        cmd.arg(arg_vec);
+
+        let result = client_clone.send_command(&cmd, None).await;
+        let client_adapter = unsafe { Box::leak(Box::from_raw(ptr_address as *mut ClientAdapter)) };
+        let value = match result {
+            Ok(value) => value,
+            Err(err) => {
+                println!(" === err {:?}", err);
+                let message = errors::error_message(&err);
+                let error_type = errors::error_type(&err);
+
+                let c_err_str = CString::into_raw(CString::new(message).unwrap());
+                unsafe { (client_adapter.failure_callback)(channel, c_err_str, error_type) };
+                return;
+            }
+        };
+
+        let mut command_response = CommandResponse {
+            int_value: 0,
+            float_value: 0.0,
+            bool_value: false,
+            string_value: std::ptr::null_mut(),
+            string_value_len: 0,
+            array_value: std::ptr::null_mut(),
+            array_elements_len: std::ptr::null_mut(),
+            array_value_len: 0,
+        };
+        let result: RedisResult<Option<CommandResponse>> = match value {
+            Value::Nil => Ok(None),
+            Value::SimpleString(text) => {
+                let vec = text.chars().map(|b| b as c_char).collect::<Vec<_>>();
+                let (vec_ptr, len) = convert_vec_to_pointer(vec);
+                command_response.string_value = vec_ptr;
+                command_response.string_value_len = len;
+                Ok(Some(command_response))
+            }
+            Value::BulkString(text) => {
+                let vec = text.iter().map(|b| *b as c_char).collect::<Vec<_>>();
+                let (vec_ptr, len) = convert_vec_to_pointer(vec);
+                command_response.string_value = vec_ptr;
+                command_response.string_value_len = len;
+                Ok(Some(command_response))
+            }
+            Value::VerbatimString { format: _, text } => {
+                let vec = text.chars().map(|b| b as c_char).collect::<Vec<_>>();
+                let (vec_ptr, len) = convert_vec_to_pointer(vec);
+                command_response.string_value = vec_ptr;
+                command_response.string_value_len = len;
+                Ok(Some(command_response))
+            }
+            Value::Okay => {
+                let vec = "OK".chars().map(|b| b as c_char).collect::<Vec<_>>();
+                let (vec_ptr, len) = convert_vec_to_pointer(vec);
+                command_response.string_value = vec_ptr;
+                command_response.string_value_len = len;
+                Ok(Some(command_response))
+            }
+            _ => todo!(),
```

Comment:
??

Reply:
The other return types of Value will be handled as we add support for more commands. It is left as TODO here.